### DISCUSSION TITLE
Unpredictable request to the server

### DISCUSSION THREAD

**Discussion by Sans84** [livewire/livewire#7101] [1. Help]
I decided to use AlpineJs and Livewire when creating components ([manual](https://livewire.laravel.com/docs/forms#custom-form-controls))

So, I created a multi-select component and placed it on a full-page component.

Full page component
```
<?php

namespace App\Livewire;

use Livewire\Attributes\Layout;
use Livewire\Attributes\On;
use Livewire\Attributes\Title;
use Livewire\Component;

#[Layout('layouts.dashboard')]
#[Title('Test')]
class TestView extends Component
{
    public array $test = ['v1', 'v2', 'v3', 'v4'];
   
    public function render()
    {
        return view('livewire.test-view');
    }
}
```

Blade `livewire.test-view`
```
<div>
    <x-test-multiselect wire:model.live="test" />
</div>
```

Component `test-multiselect`
```
    <div wire:ignore x-data="{values: []}" x-modelable="values" {{ $attributes  }}>
        <select x-model="values" multiple>
            <option value="v1">v1</option>
            <option value="v2">v2</option>
            <option value="v3">v3</option>
            <option value="v4">v4</option>
        </select>
    </div>
```

We have the following rendering
<img width="61" alt="image" src="https://github.com/livewire/livewire/assets/18501836/a08df6ed-6ced-4a39-974a-ba90300029ec">

deselect one option and two requests are sent to the server
<img width="1410" alt="image" src="https://github.com/livewire/livewire/assets/18501836/6cbcda09-411c-4966-b139-29c22f2f382e">
<img width="1162" alt="image" src="https://github.com/livewire/livewire/assets/18501836/be207675-5d44-445a-8132-e24362e906b7">

Deselect in the following order - v2, v4, v1
The selection remains selected on v3, but the property is set to ['v1', 'v3', 'v4'] Why?

```
public function updatedTest($value, $key)
    {
        \Log::info('updatedTest: ', [$this->test, $value, $key]);
    }

local.INFO: updatedTest:  [["v1","v3","v3","v4"],"v3","1"] // remove v2
local.INFO: updatedTest:  [["v1","v3","v4","v4"],"v4","2"] // remove v4
local.INFO: updatedTest:  [["v1","v3","v4"],"__rm__","3"] // remove v1
```

**Comment by seabasss** [comment]
Same issue with `__rm__` here. Any ideas how to solve it?

**Comment by RenoLooijmans** [comment]
Bug is still present... Please see
https://github.com/livewire/livewire/pull/8235

and for possible workaround: https://github.com/livewire/livewire/discussions/6665
